### DISCUSSION TITLE
ch4/ch4-04

### DISCUSSION THREAD

**Discussion by giscus** [gopl-zh/gopl-zh.github.com#183] [General]
# ch4/ch4-04

<The Go Programming Language>中文版

https://golang-china.github.io/gopl-zh/ch4/ch4-04.html

**Comment by soondubu137** [comment]
4.4.3节最后关于可见性的讨论中，
> “但是在包外部，因为circle和point没有导出，不能访问它们的成员，因此简短的匿名成员访问语法也是禁止的。”

这句我觉得翻译的不太对/好。原文是这样的：
> Had they been unexported (`point` and `circle`), we could still use the shorthand form, but the explicit long form shown in the comment would be forbidden outside the declaring package because `circle` and `point` would be inaccessible.

即使`point`和`circle`不导出，我们仍然可以使用简短形式来访问`w.X`，但是像注释中的那种长形式（`w.circle.point.X`）会在外部包中被禁止，因为你无法访问到`circle`和`point`。

**Reply by a-firefly** [reply]
实测发现确实如此：即使point和circle不导出，我们仍然可以使用简短形式来访问w.X，但是像注释中的那种长形式（w.circle.point.X）会在外部包中被禁止，因为你无法访问到circle和point。

**Comment by Apahno** [comment]
打卡

**Comment by Phinome** [comment]
dilbert.Salary -= 5000 // demoted, for writing too few lines of code
真实的贷款打工

**Comment by ww1436354649** [comment]
struct里的匿名成员不是一个好的设计
